pkg/utils/hashlib/hasher.go

```diff
@@ -0,0 +1,13 @@
+package hashlib
+
+// Hash contains all supported hash formats.
+// Add additional hash types e.g. [20]byte as needed here.
+type Hash interface {
+	[32]byte
+}
+
+type Ctx[H Hash] interface {
```

Comment:
What is the significance of `Ctx` here?
Would something simple and direct be fitting, like `Hasher`?
```suggestion
type Hasher[H Hash] interface {
```